x-pack/plugins/ml/server/lib/capabilities/check_capabilities.ts

```diff
@@ -47,3 +49,20 @@ function disableAdminPrivileges(capabilities: MlCapabilities) {
   capabilities.canCreateAnnotation = false;
   capabilities.canDeleteAnnotation = false;
 }
+
+export type HasMlCapabilities = (capabilities: MlCapabilitiesKey[]) => void;
+
+export function hasMlCapabilitiesProvider(resolveMlCapabilities: ResolveMlCapabilities) {
+  return (request: KibanaRequest) => {
```

Comment:
Is there a way you could make use of the `HasMlCapabilities` type to enforce it? Looking at the code like this I'm unsure why the type `HasMlCapabilities` is a function without `async` and the return function in the provider uses `async`.

Reply:
`HasMlCapabilities` should really return `Promise<void>` here. i'll update it.
I'll also add `HasMlCapabilities` as the return type of this returned function to make it more explicit.